text/0000-layer-metadata-files.md

```diff
@@ -0,0 +1,211 @@
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposal is as follows -
+
+The current folder structure is -
+
+```
+/cnb (provided by builder image)
+├── buildpacks
+│   └── <buildpacks>
+├── lifecycle
+│   └── <creator/builder/etc>
+├── order.toml
+└── stack.toml
+/platform (configurable with `CNB_PLATFORM_DIR`)
+├── env
+└   └── <env>
+/layers (configurable with `CNB_LAYERS_DIR` or -layers)
+├── analyzed.toml
+├── group.toml
+├── plan.toml
+├── order.toml (written by a platform, optionally - prior to `lifecycle` execution)
+├── project-metadata.toml (written by a platform, optionally - prior to `lifecycle` execution, configurable with `CNB_PROJECT_METADATA_PATH`)
+├── report.toml
+├── <escaped_buildpack_id> (buildpack arg `$1`)
+│   ├── build.toml
+│   ├── launch.toml
+│   ├── store.toml
+│   ├── <layer>.toml
+│   └── <layer>
+└── config
+    └── metadata.toml
+/workspace (configurable with `CNB_APP_DIR ` or -app)
+└── <app source code to transform>
+```
+
+This proposal is to change this folder structure to -
+
+```
+/cnb (provided by builder image)
+├── buildpacks
+│   └── <buildpacks>
+├── lifecycle
+│   └── <creator/builder/etc>
+├── order.toml
+└── stack.toml
+/platform (configurable with `CNB_PLATFORM_DIR`)
+├── env
+└   └── <env>
+/workspace (configurable with `CNB_WORKSPACE_DIR`)
+├── app (configurable with `CNB_APP_DIR ` or -app)
+│    └── <app source code to transform>
+├── layers (configurable with `CNB_LAYERS_DIR` or -layers)
+│   └── new-buildpack (`CNB_BUILDPACK_LAYERS_DIR`)
+│       ├── <layer>.toml
+│       └── <layer>
+│   └── old-buildpack (buildpack arg `$1`)
+│       ├── build.toml
+│       ├── launch.toml
+│       ├── store.toml
+│       ├── <layer>.toml
+│       └── <layer>
+├── config
+│   └── layers (configurable with `CNB_LAYERS_CONFIG_DIR`)
+│       ├── metadata.toml
+│       └── new-buildpack (`CNB_BUILDPACK_CONFIG_DIR`)
+│        ├── build.toml
+│        ├── launch.toml
+│        └── store.toml
+│   └── lifecycle (configurable with `CNB_LIFECYCLE_CONFIG_DIR`)
+│       ├── analyzed.toml
+│       ├── group.toml
+│       ├── plan.toml
+│       └── report.toml
+│   └── platform (configurable with `CNB_PLATFORM_CONFIG_DIR`)
+│       ├── order.toml (written by a platform, optionally - prior to `lifecycle` execution)
+└       └── project-metadata.toml (written by a platform, optionally - prior to `lifecycle` execution, configurable with `CNB_PROJECT_METADATA_PATH`)
```

Comment:
Would this make sense to go in the platform directory? In talking through https://github.com/buildpacks/pack/issues/1221 it seems like it could be helpful for the platform to mount a directory where it could provide configuration and receive output.